broker-daemon/block-order-worker/index.spec.js

```diff
@@ -2115,22 +2118,77 @@ describe('BlockOrderWorker', () => {
     })
 
     describe('reject event', () => {
+      let workBlockOrderStub
       let failBlockOrderStub
+      let relayerIsAvailableStub
+      let getBlockOrderStub
+      let updatedBlockOrder
       let rejectListener
+      let retry
 
       beforeEach(() => {
+        workBlockOrderStub = sinon.stub().resolves()
+        worker.workBlockOrder = workBlockOrderStub
+
         failBlockOrderStub = sinon.stub().resolves(true)
         worker.failBlockOrder = failBlockOrderStub
+
+        relayerIsAvailableStub = sinon.stub().resolves(true)
+        worker.relayerIsAvailable = relayerIsAvailableStub
+
+        updatedBlockOrder = blockOrder
+        getBlockOrderStub = sinon.stub().resolves(updatedBlockOrder)
+        worker.getBlockOrder = getBlockOrderStub
+
+        retry = sinon.stub()
+        BlockOrderWorker.__set__('retry', retry)
+
         rejectListener = onceStub.withArgs('reject').args[0][1]
       })
 
       it('registers an event on an order for order rejection', async () => {
         expect(onceStub).to.have.been.calledWith('reject', sinon.match.func)
       })
 
-      it('fails a block order if the call is rejected', async () => {
+      it('retries the block order when it should', async () => {
```

Comment:
would add the situation here to be more specific. Example `retries the block order when the relayer is available`